src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -54,14 +56,31 @@ export class MetadataBlockComponent extends BaseProcessingComponent implements O
             this.loadMetadataBlock(),
             this.loadHistory(),
             this.appDatasetSubsService.onDatasetHistoryChanges.subscribe((result: DatasetHistoryUpdate) => {
-                this.datasetHistoryUpdate = result;
+                this.mapHistory.set(result.pageInfo.currentPage, result);
+                if (result.pageInfo.hasNextPage && result.pageInfo.totalPages !== this.mapHistory.size) {
+                    this.loadHistory(result.pageInfo.currentPage + 1);
+                }
+                const firstPageHistory = this.mapHistory.get(0);
+                if (firstPageHistory) {
+                    this.datasetHistoryUpdate = firstPageHistory;
+                }
                 this.cdr.detectChanges();
             }),
         );
     }
 
+    public get allHistory(): MetadataBlockFragment[] {
+        return Array.from(this.mapHistory.values())
+            .map((item) => item.history)
+            .reduce((acc, cur) => [...acc, ...cur], []);
```

Comment:
Looks N^2 to me by complexity - you are creating lists and copying all elements + another page's elements on each step.

Consider implementing something like a _Composite Iterator_. 
Or maybe a wrapper interface around all arrays without copying them and access to a node by index, which would map to one of the arrays depending on index / pageSize.